libs/product/src/actions/configurable-product.actions.ts

```diff
@@ -0,0 +1,55 @@
+import { Action } from '@ngrx/store';
+
+import { DaffConfigurableProduct } from '../models/configurable-product';
+
+/**
+ * Action types for Configurable Product Actions.
+ */
+export enum DaffConfigurableProductActionTypes {
+	ConfigurableProductApplyAttributeAction = '[Configurable Product] Configurable Product Apply Attribute Action',
+	ConfigurableProductRemoveAttributeAction = '[Configurable Product] Configurable Product Remove Attribute Action',
+	ConfigurableProductToggleAttributeAction = '[Configurable Product] Configurable Product Toggle Attribute Action',
+}
+
+/**
+ * Applies an attribute to a particular configurable product.
+ * 
+ * @param id - Id of the Configurable Product
+ * @param attributeId - Id of the attribute to be applied
+ * @param attributeValue - Value of the attribute to be applied
+ */
+export class DaffConfigurableProductApplyAttribute<T extends DaffConfigurableProduct> implements Action {
+  readonly type = DaffConfigurableProductActionTypes.ConfigurableProductApplyAttributeAction;
+
+  constructor(public id: T['id'], public attributeId: string, public attributeValue: string) {}
+}
+
+/**
+ * Removes an applied attribute from a particular configurable product.
+ * 
+ * @param id - Id of the Configurable Product
+ * @param attributeId - Id of the attribute to be removed
+ */
+export class DaffConfigurableProductRemoveAttribute<T extends DaffConfigurableProduct> implements Action {
+  readonly type = DaffConfigurableProductActionTypes.ConfigurableProductRemoveAttributeAction;
+
+  constructor(public id: T['id'], public attributeId: string) {}
+}
+
+/**
+ * Toggles an attribute of a particular configurable product.
+ * 
+ * @param id - Id of the Configurable Product
+ * @param attributeId - Id of the attribute to be toggled
+ * @param attributeValue - Value of the attribute to be toggled
+ */
+export class DaffConfigurableProductToggleAttribute<T extends DaffConfigurableProduct> implements Action {
+  readonly type = DaffConfigurableProductActionTypes.ConfigurableProductToggleAttributeAction;
+
+  constructor(public id: T['id'], public attributeId: string, public attributeValue: string) {}
+}
```

Comment:
I'm not sure that a toggle action makes sense for values that aren't boolean. LMK what you think; I might be misunderstanding something.

Reply:
oh, you're right. I'll take another look at it

Reply:
I changed how the reducer handles the action. Basically, I still want to allow for a toggle-type action, so that the app dev can just use that instead of the apply/remove actions. But I hadn't thought of the usecase (which would happen all the time) for when the color blue was selected and the new selection is the color red. So now the app dev can just use the toggle action for any time the user clicks on a new configurable attribute. So to be clear on the use cases:
1. Configurable T-shirt has the attribute color set to blue. User clicks on the blue swatch again to deselect the attribute color. 
2. Configurable T-shirt has no attribute color selected. User clicks on the blue swatch to select the attribute color.
3. Configurable T-shirt has the attribute color set to blue. User clicks on the red swatch to change their selection of the color attribute to red. (this is what previously wouldn't have worked)

Reply:
Alright lgtm. I'm convinced that this is a use case that we want to support.